sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -103,6 +107,13 @@ def fit(self, X, y):
         acc_prediction_time = 0.
         X, y = check_X_y(X, y, dtype=[X_DTYPE], force_all_finite=False)
         y = self._encode_y(y)
+        check_consistent_length(X, y)
+        # Do not create unit sample weights by default to later skip some
+        # computation
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X,
+                                                 dtype=np.float64)
```

Comment:
Not related to this PR but I'm tempted to add a `return_ones` option to  `_check_sample_weight` because this is annoying me a little

Reply:
I agree, Added the parameter, need to make our sample_weight checks consistent in another PR.